Service/Microsoft.Health.DeIdentification.Local/InMemoryQueueClient.cs

```diff
@@ -1,37 +1,106 @@
 ﻿using Microsoft.Health.JobManagement;
+using System.Collections.ObjectModel;
 
 namespace Microsoft.Health.DeIdentification.Local
 {
     public class InMemoryQueueClient : IQueueClient
     {
+        private int largestId = 1;
+        private List<JobInfo> jobInfos = new List<JobInfo>();
+        public List<JobInfo> JobInfos
+        {
+            get { return jobInfos; } 
+        }
+
         public Task CancelJobByGroupIdAsync(byte queueType, long groupId, CancellationToken cancellationToken)
         {
             throw new NotImplementedException();
         }
 
         public Task CancelJobByIdAsync(byte queueType, long jobId, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            foreach (JobInfo jobInfo in jobInfos.Where(t => t.Id == jobId))
+            {
+                if (jobInfo.Status == JobStatus.Created)
+                {
+                    jobInfo.Status = JobStatus.Cancelled;
+                }
+
+                if (jobInfo.Status == JobStatus.Running)
+                {
+                    jobInfo.CancelRequested = true;
```

Comment:
the cancelrequest needs always to be set to true.

Reply:
I will modify it